src/main/java/de/nplay/moderationbot/config/bot/type/BotConfigType.java

```diff
@@ -0,0 +1,7 @@
+package de.nplay.moderationbot.config.bot.type;
+
+import java.util.function.Function;
+
+public record BotConfigType(String displayName, Function<String, Boolean> validator,
```

Comment:
formatter can be replaced with `String format, String value` signature, `#format(...)` is then called in the constructor